dask-image v2023.8.1

conda-forge-admin opened the pull request:
Hi! This is the friendly automated conda-forge-webservice.

I've started a version update as instructed in #13.

I'm currently searching for new versions and will update this PR shortly if I find one! Thank you for waiting!


Fixes #13
Closes #15

conda-forge-webservices[bot] committed:
dummy commit for rerendering

Commit:
ENH updated version to 2023.8.0

Commit:
MNT: Re-rendered with conda-build 3.26.0, conda-smithy 3.24.1, and conda-forge-pinning 2023.08.03.14.27.32

m-albert committed:
Updated requirements for 2023.08.0

m-albert committed:
Removed scikit-image from test requirements

jakirkham commented on recipe/meta.yaml:
```diff
@@ -19,16 +19,17 @@ build:
   host:
     - pip
-    - python >=3.8
+    - python >=3.9
```
As we specify [Python 3.8+ in `setup.py`]( https://github.com/dask/dask-image/blob/v2023.08.0/setup.py#L64 ) should this match?

```suggestion
    - python >=3.8
```

m-albert replied:
Good catch!

Since python 3.8 is officially dropped (https://github.com/dask/dask-image/pull/315), probably this should be corrected in `setup.py` (to python 3.9+)?

jakirkham replied:
Maybe we should do a patch release then? The `whl`s and `sdist` will be installable on Python 3.8 given this constraint

m-albert replied:
With patch release do you mean a 2023.08.1 right after this one? That'd make sense to me.

(This comment had got stuck behind the review submission earlier)

jakirkham replied:
Submitted PR ( https://github.com/dask/dask-image/pull/336 ) to bump that version

Since the `whl`s and `sdist` are allowing Python 3.8, would it make sense to stick with Python 3.8 one more release?

Alternatively we can do a patch release with the PR above

Think it would be good if the packages are installed in a consistent manner across distributions (as we saw today with Dask & Arrow inconsistency can lead to surprising behavior for users)

m-albert replied:
> Since the whls and sdist are allowing Python 3.8, would it make sense to stick with Python 3.8 one more release?

https://github.com/dask/dask-image/pull/315 dropped 3.8 not because of a specific incompatibility, but the NEP29 recommendation to drop support for python 3.8 after 04/2023. So in principle supporting 3.8 should be okay. 

Inconsistencies/disadvantages that come to mind would be:
- testing was done for 3.9+
- the release notes mention dropping 3.8 as a highlight 😅
- `setup.py` has 3.9+ classifiers

jakirkham replied:
Yeah if we would rather do a patch release with PR ( https://github.com/dask/dask-image/pull/336 ) and just bump this PR to 2023.8.1, that seems reasonable to me

m-albert replied:
Oh I see. Just to clarify that I understood you correctly. In terms of concrete steps, your suggestion would be to:
1) Create a 2023.8.1 release on github.
2) Upload to pypi
3) Bump this PR by modifying
  - `set version = "2023.8.0"` to `set version = "2023.8.1"`
  - `set sha256 = "f68064320858ca687d05201277fb9937f47ff0cfc97255689ed8ad558a2a40b2"` to ...?

This would lead to 2023.08.0 being available only on pypi and not conda.

jakirkham replied:
Yeah we can also yank 2023.8.0 on PyPI so it wouldn't show up on either

m-albert replied:
Ah yes. I'm thinking what that'd mean for the release notes. We could probably just include the same notes for 2023.08.0 with a comment saying they're shown again since 2023.08.1 is a patch release.

m-albert replied:
How about deleting the v2023.08.0
- github release
- git tag and
- pypi release files

and release it under the same version again (including https://github.com/dask/dask-image/pull/336). That's probably a bad option?

jakirkham replied:
Yeah users may already have the release now in some form. Removal could cause confusion

Adding a new release is typically the preferred path

jakirkham replied:
FWIW here's how one project [updates release notes for these kinds of issues]( https://github.com/nicolargo/glances/blob/407a3cde210013faca35af0ff7bd652787d8f86f/NEWS.rst#version-3403 )

m-albert replied:
I see, that makes sense! Guess I was wondering about the release notes for 2023.08.0 missing on pypi if we yank it, but people could just refer to the complete version list on github.

Thank you very much John for all your efforts here and patience with me :)

I'll need to get some sleep now and tomorrow morning will create the patch release and work on this PR as discussed.

jakirkham replied:
Of course! Happy to help 🙂

Sorry this has been a bit of a bumpy ride here

Hopefully we can learn from these issues and make the experience smoother in the future

GenevieveBuckley replied:
I think this is a good plan.

jakirkham commented on recipe/meta.yaml:
```diff
@@ -19,16 +19,17 @@ build:
-    - python >=3.8
-    - dask >=2021.10.0
-    - numpy >=1.11.3
+    - python >=3.9
```
Same question as above

```suggestion
    - python >=3.8
```

jakirkham reviewed (commented):
Thanks Marvin! 🙏

Had a couple questions below

jakirkham commented:
For reference the discussion regarding the CI test failures here is occurring upstream: https://github.com/dask/dask-image/issues/330#issuecomment-1664286135

jakirkham committed:
Disable Dask's Arrow-based string support

The new Arrow-based strings that Dask uses appear to cause dask-image
test failures. So just disable them for now. Also link to an upstream
tracking issue for reference.

jakirkham commented on recipe/meta.yaml:
```diff
@@ -39,9 +40,12 @@ test:
     - pytest >=3.0.5
-    - scikit-image >=0.12.3,<0.17.0
 
   commands:
+    # Disable newer Arrow strings with dask-image.
+    # xref: https://github.com/dask/dask-image/issues/335
+    - export DASK_DATAFRAME__CONVERT_STRING="False"  # [unix]
+    - set "DASK_DATAFRAME__CONVERT_STRING=False"     # [win]
     - pytest
```
As noted above, using the newer Arrow-based strings added in Dask are causing test failures in dask-image. So just disable this option until we can figure out how to fix them

Raised upstream issue (  https://github.com/dask/dask-image/issues/335 ) to track resolution

<hr>

The Dask config docs show [`dataframe.convert-string`]( https://docs.dask.org/en/stable/configuration.html#dataframe.convert-string ) as the option to use for configuring Arrow-based strings in Dask

Based on offline testing, this seems to be the right environment variable to set to affect the `dask.config`. Would note the default value is `None`, which IIUC means it will use Arrow-based strings if Arrow is available. So we can see that it is now `False`, which is what we need to disable them completely

```bash
$ DASK_DATAFRAME__CONVERT_STRING="False" python -c "import dask; print(dask.config.get('dataframe.convert-string'))"
False
```

jakirkham replied:
Appears to be working! 🎉

m-albert replied:
It does! The failing test passes now 🎉🚀

m-albert reviewed (commented):
Thanks so much for fixing this @jakirkham ! Added some comments.

m-albert committed:
Update recipe/meta.yaml

Ensure consistency with v2023.08.0 on pypi and therefore also support python 3.8 here.

Co-authored-by: jakirkham <jakirkham@gmail.com>

GenevieveBuckley reviewed (approved):
This all looks pretty good, and I approve the plan for a patch v2023.8.1 to get rid of the last references to python 3.8. Sorry we didn't catch them all earlier!

Definitely don't worry at all about whether the release history looks slightly messy, nobody will care about that. 

I agree it's a good idea to yank v2023.8.0 from pypi in favour of 2023.8.1. But again, it seems very unlikely that anybody is going to actually end up installing it with python 3.8 and have problems, so don't worry too much about it.

m-albert committed:
Bump release version to 2023.8.1

m-albert committed:
Set host requirement to python>=3.9

m-albert committed:
Update hash

m-albert commented:
Thanks for the change of PR title @jakirkham, that clarifies things :)

All seems good, merging now! Thanks @jakirkham and @GenevieveBuckley 🙏

EDIT: Accidentally clicked on "close with comment" on this one 🙈 Not a problem though 😅.

jakirkham commented:
Thanks Marvin! 🥳

If you have a moment to write your thoughts on how this release went and what we could improve, that would be really helpful
